test/features/events.js

```diff
@@ -0,0 +1,312 @@
+'use strict';
+
+var Event2 = require('../../api/models/Event');
+
+Feature('Check In Event', () => {
+
+  Scenario('Non-Existent Event should be Created', () => {
+
+    var date = new Date(923423234);
+    var dateString = date.toISOString();
+    var payload = {
+      person_id: 123,
+      cid_id: 999,
+      operation: 'check in',
+      timestamp: dateString,
+      gender: 'm',
+      nationality: 'gbr',
+      centre: 'bigone'
+    };
+
+    before(function () {
+      global.testConfig.initializeBarrelsFixtures = false;
+    });
+
+    after(function () {
+      global.testConfig.initializeBarrelsFixtures = true;
+    });
+
+    Given('a detainee event with timestamp "' + dateString + '" has not already occurred', () =>
+
+      global.initializeBarrelsFixtures()
+        .then(() =>
+          Event.find({
+            where: {
+              timestamp: dateString,
+              operation: 'check in'
+            }
+          }).then((models) =>
+            expect(models.length).to.equal(0)
+          )
+        )
+    );
+
+    When('a valid detainee event with timestamp "' + dateString + '" occurs', () =>
+
+      request(sails.hooks.http.app)
+        .post('/irc_entry/event')
+        .send(payload)
+        .expect(201)
+    );
+
+    Then('an event with specific timestamp "' + dateString + '" is created from the detainee event received', () =>
+
+      Event.find({
+        where: {
+          timestamp: dateString,
+          operation: 'check in',
+          centre: 1
+        }
+      }).then((models) => expect(models.length).to.equal(1))
+    );
+
+  });
+
+  Scenario('Event should be created', () => {
+
+    var date = new Date();
+    var dateString = date.toISOString();
+    var payload = {
+      person_id: 123,
+      cid_id: 100,
+      operation: 'check in',
+      timestamp: dateString,
+      gender: 'm',
+      nationality: 'gbr',
+      centre: 'bigone'
+    };
+
+    before(function () {
+      global.testConfig.initializeBarrelsFixtures = false;
+    });
+
+    after(function () {
+      global.testConfig.initializeBarrelsFixtures = true;
+    });
+
+    var createEvent = function () {
+      return request(sails.hooks.http.app)
+        .post('/irc_entry/event')
+        .send(payload)
+        .expect(201);
+    }
+
+    Given('a detainee event with timestamp "' + dateString + '" has already occurred', () =>
```

Comment:
having a changing date in the test description will make it impossible to track failing tests either pick a fixed date or say 'now' in the given text